src/main/java/reactor/kafka/receiver/internals/DefaultKafkaReceiver.java

```diff
@@ -334,6 +346,11 @@ private void cancel(boolean async) {
                     consumer.wakeup();
                     CloseEvent closeEvent = new CloseEvent(receiverOptions.closeTimeout());
                     if (async) {
+                        if (Thread.currentThread().getName().startsWith("reactive-kafka-")) {
```

Comment:
since this is an important check, I would suggest reusing the constant between this place and the thread creator

Reply:
removed it for now. Will fix in the #38